lib/Serialization/SerializedModuleLoader.cpp

```diff
@@ -409,16 +405,20 @@ bool SerializedModuleLoader::maybeDiagnoseTargetMismatch(
       return false;
     StringRef filePath = directoryIterator->path();
     StringRef extension = llvm::sys::path::extension(filePath);
-    if (file_types::lookupTypeForExtension(extension) ==
-          file_types::TY_SwiftModuleFile) {
+    auto fileType = file_types::lookupTypeForExtension(extension);
+    // We also check for interfaces here, because the SerializedModuleLoader
+    // is invoked after the ModuleInterfaceLoader; if the ModuleInterfaceLoader
+    // handled interfaces separately, we could get duplicate diagnostics.
+    if (fileType == file_types::TY_SwiftModuleFile
+        || fileType == file_types::TY_SwiftModuleInterfaceFile) {
```

Comment:
Nice catch.

Nit: Should `||` be on the line before?

Reply:
For things we don't have written out (e.g. LLVM coding standards doesn't mention it) and don't seem to have a consistent style in the code (this file has both styles in use), I fallback (heh) to [PEP-8](https://www.python.org/dev/peps/pep-0008/#should-a-line-break-before-or-after-a-binary-operator).

For example, here's some of the code you added recently in the same file:

```
    if (firstAbsoluteBaseName
        && maybeDiagnoseTargetMismatch(moduleID.Loc, moduleName,
                                       *firstAbsoluteBaseName)) {
```

Earlier, it was

```
    if (maybeDiagnoseTargetMismatch(moduleID.Loc, moduleName,
                                    primaryTargetSpecificName, currPath)) {
```

There's also examples of it being formatted with the operator at the end in the same file. 🤷